pkg/apk/installed.go

```diff
@@ -34,7 +34,7 @@ import (
 
 type InstalledPackage struct {
 	Package
-	Files []*tar.Header
+	Files []tar.Header
```

Comment:
For a big enough package, the size of these `tar.Header`s starts to add up and we end up spending a lot of time copying them when growing `Files`. I'd like to benchmark this a little bit to make sure we don't see a performance regression. We might want to adapt everything else to use pointers.

Reply:
I did a simple test with 100k files in installed file:

Methodology:

1. Generate Apkindex file with:

```bash
#!/bin/bash

echo P:package
echo T:A sample package
echo F:dir

for i in $(seq 1 100000); do
	echo "R:file_$i"
done

echo
```

2. Run the test

```go
func TestBenchmark(t *testing.T) {
	for i := 0; i < 10; i++ {
		f, err := os.Open("installed_many_files")
		require.NoError(t, err)
		start := time.Now()
		pkg, err := ParseInstalled(f)
		fmt.Println(time.Since(start))
		require.NoError(t, err)
		require.Len(t, pkg[0].Files, 100001)
	}
}
```

**Results**

With values:

```
=== RUN   TestBenchmark
30.626125ms
23.098833ms
25.923584ms
24.632208ms
25.329541ms
24.9935ms
25.655959ms
25.694375ms
25.335208ms
24.423292ms
--- PASS: TestBenchmark (0.26s)
PASS
```

With pointers:

```
=== RUN   TestBenchmark
17.986708ms
15.816167ms
17.7ms
15.778584ms
15.979667ms
17.57225ms
15.229ms
16.453792ms
16.038333ms
16.060916ms
--- PASS: TestBenchmark (0.17s)
PASS
```

Reply:
Thanks for doing the legwork, this change seems fine.